core/domain/collection_domain.py

```diff
@@ -1102,6 +1132,64 @@ def delete_skill(self, skill_id):
 
         del self.skills[skill_id]
 
+    def add_question_id_to_skill(self, skill_name, question_id):
+        """Adds the question id to the question list of the appropriate skill.
+
+        Args:
+            skill_name: str. The name of the skill.
+            question_id: str. The id of the question.
+
+        Raises:
+            Exception: question_id is already present in skill.
+        """
+        index = 0
+        skill_index = None
+        for _, skill in self.skills.iteritems():
+            if skill.name == skill_name:
+                skill_index = index
+            index += 1
+        if skill_index is not None:
+            skill_id = CollectionSkill.get_skill_id_from_index(skill_index)
+            question_ids = self.skills[skill_id].question_ids
+            if question_id not in question_ids:
+                self.skills[skill_id].question_ids.append(
+                    question_id)
+            else:
+                raise Exception(
+                    '%s is already present in %s' % (
+                        self.question_id, skill_name))
+
+    def remove_question_id_from_skill(self, skill_id, question_id):
+        """Removes question id from the question list of the appropriate skill.
+
+        Args:
+            skill_id: str. The id of the skill.
+            question_id: str. The id of the question.
+
+        Raises:
+            Exception: question_id is not present in the skill.
+        """
+        if question_id not in self.skills[skill_id].question_ids:
+            raise Exception(
+                '%s is not present in %s' % (
+                    question_id, self.skills[skill_id].name))
+        else:
+            self.skills[skill_id].question_ids.remove(question_id)
+
+    def get_acquired_skill_ids_from_exploration_ids(self, exploration_ids):
+        """Returns a list of skill ids acquired by completing the explorations
```

Comment:
... completing the given explorations in the collection.

Reply:
Updated.